test/role.test.js

```diff
@@ -489,48 +491,246 @@ describe('role model', function() {
+      {username: 'receiver', email: 'receiver@example.com', password: 'pass'},
+    ];
+
+    it('supports legacy behavior with no relation via property "userId"', function() {
+      var user;
+      var Album = app.registry.createModel('Album', {
+        name: String,
+        userId: Number,
+      });
+      app.model(Album, {dataSource: 'db'});
 
-    var user;
-    return User.create({email: 'test@example.com', password: 'pass'})
+      return User.create({email: 'test@example.com', password: 'pass'})
       .then(u => {
         user = u;
         return Album.create({name: 'Album 1', userId: user.id});
       })
       .then(album => {
         return Role.isInRole(Role.OWNER, {
-          principalType: ACL.USER, principalId: user.id,
-          model: Album, id: album.id,
+          principalType: ACL.USER,
+          principalId: user.id,
+          model: Album,
+          id: album.id,
         });
       })
       .then(isInRole => expect(isInRole).to.be.true());
-  });
-
-  it('resolves OWNER via "owner" property with no relation', function() {
-    var Album = app.registry.createModel('Album', {
-      name: String,
-      owner: Number,
     });
-    app.model(Album, {dataSource: 'db'});
 
-    var user;
-    return User.create({email: 'test@example.com', password: 'pass'})
+    it('supports legacy behavior with no relation via property "owner"', function() {
+      var user;
+      var Album = app.registry.createModel('Album', {
+        name: String,
+        owner: Number,
+      });
+      app.model(Album, {dataSource: 'db'});
+
+      return User.create({email: 'test@example.com', password: 'pass'})
       .then(u => {
         user = u;
         return Album.create({name: 'Album 1', owner: user.id});
       })
       .then(album => {
         return Role.isInRole(Role.OWNER, {
-          principalType: ACL.USER, principalId: user.id,
-          model: Album, id: album.id,
+          principalType: ACL.USER,
+          principalId: user.id,
+          model: Album,
+          id: album.id,
         });
       })
       .then(isInRole => expect(isInRole).to.be.true());
+    });
+
+    it('supports legacy behavior via a belongsTo relation', function() {
+      // passing no options will result calling
+      // the legacy $owner role resolver behavior
+      var Message = createModelWithOptions('ModelWithNoOptions');
+
+      return givenUsers()
+      .then(() => {
+        var messages = [
+          {content: 'firstMessage', senderId: sender.id},
+          {content: 'secondMessage', receiverId: receiver.id},
+          {content: 'thirdMessage'},
+        ];
+        return Promise.map(messages, msg => {
+          return Message.create(msg);
+        });
+      })
+      .then(messages => {
+        return Promise.all([
+          isOwnerForMessage(sender, messages[0]),
+          isOwnerForMessage(receiver, messages[1]),
+          isOwnerForMessage(receiver, messages[2]),
```

Comment:
Should we check `isOwnerForMessage(sender, messages[2])` too?

I find it difficult to understand what is the rationale for choosing exactly these three combinations. The tests would be much easier to read if each test case was in its own, well-named test.

```js
describe('legacy behavior via a belongsTo relation', () => {
  let message;
  beforeEach(givenModelWithSenderReceiverRelations);
  beforeEach(givenUsers);

  it('resolves true when the first belongsTo relation matches', () => {
    return Message.create({content: 'firstMessage', senderId: sender.id})
      .then(m => isOwnerForMessage(sender, m))
      .then(result => expect(result.isOwner).to.be.true());
    }
  });
});
```

I think the total number of source code lines may be even smaller using my approach.